### PR TITLE
Updated code and formatting issues fixed

### PR DESCRIPTION
**Pull request by trapacska**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Place your settings in this file to overwrite default and user settings.
```
Comment (viktorbenei) [inline .vscode/settings.json]: please `.gitignore` the `.vscode` dir ;)

File: bitrise.yml [hunk 2]
```diff
@@ -11,6 +11,9 @@ workflows:
   # ----------------------------------------------------------------
   # --- workflow to test this step
   test:
+    before_run:
+    - godeps-update
```
Comment (viktorbenei) [inline bitrise.yml]: don't update the deps for every test!

File: step.go [hunk 3]
```diff
@@ -7,133 +7,130 @@ import (
...
-}
+		fromNameOnError:     os.Getenv("from_name_on_error"),
+		messageOnError:      os.Getenv("message_on_error"),
+		messageColorOnError: os.Getenv("message_color_on_error"),
 
-func logDetails(format string, v ...interface{}) {
-	errorMsg := fmt.Sprintf(format, v...)
-	fmt.Printf("  %s\n", errorMsg)
+		messageFormat:     os.Getenv("message_format"),
+		isBuildFailedMode: os.Getenv("STEPLIB_BUILD_STATUS"),
```
Comment (viktorbenei) [inline step.go]: We should use `BITRISE_BUILD_STATUS` instead (http://devcenter.bitrise.io/faq/available-environment-variables/)

**Comment by viktorbenei** [comment]
@godrei pls add me to this repo's bitrise.io CI project

**Comment by viktorbenei** [comment]
LGTM - wdyt @godrei ?

File: step.go [hunk 4]
```diff
@@ -7,133 +7,129 @@ import (
...
-		logFail("Failed to create request, error: %s", err)
+		log.Errorf("failed to perform request, error: %s", err)
+		os.Exit(1)
 	}
+
 	request.Header.Add("Accept", "application/json")
 	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 
 	client := &http.Client{}
 	response, requestErr := client.Do(request)
```
Comment (godrei) [inline step.go]: Could you please rename `requestErr` to `err` and handle the error here? If error is not nil, the body will be empty.

Please take a look, how we handle request in [HockeyApp step](https://github.com/bitrise-io/steps-hockeyapp-deploy/blob/master/main.go#L182)

File: step.go [hunk 5]
```diff
@@ -7,133 +7,129 @@ import (
...
 
-	defer response.Body.Close()
```
Comment (godrei) [inline step.go]: do not forget to close the body

File: step.go [hunk 6]
```diff
@@ -7,133 +7,129 @@ import (
...
 	contents, readErr := ioutil.ReadAll(response.Body)
```
Comment (godrei) [inline step.go]: `readErr` should be `err` and error should be handled here.

File: step.go [hunk 7]
```diff
@@ -142,34 +138,43 @@ func main() {
 	// Error
 	if requestErr != nil {
```
Comment (godrei) [inline step.go]: Please handle request and read error similar to HockeyApp step

**Comment by godrei** [comment]
@trapacska just one more thing, please use go-toolkit, read more in [Toolkit support (BETA) section](http://blog.bitrise.io/2016/09/13/monthly-release-of-bitrise-cli-sept.html)

Summary:
- specify package name in step.yml (https://github.com/bitrise-io/steps-slack-message/blob/master/step.yml#L31)
- update [step.sh](https://github.com/bitrise-io/steps-slack-message/blob/master/step.sh) to suport older bitrise-cli versions (without toolkit feature)
- rename step.go to main.go

**Pull request merged**